presto-main/src/main/java/io/prestosql/execution/scheduler/ScaledWriterScheduler.java

```diff
@@ -90,9 +91,10 @@ private int getNewTaskCount()
             return 1;
         }
 
-        double fullTasks = sourceTasksProvider.get().stream()
-                .filter(task -> !task.getState().isDone())
-                .map(TaskStatus::isOutputBufferOverutilized)
+        // only consider source tasks which have started running and have an initialized output buffer
+        double sourcesBufferUtilizationAvg = sourceTasksProvider.get().stream()
+                .filter(task -> task.getState() == TaskState.RUNNING && task.getOutputBufferUtilization().isPresent())
```

Comment:
Could we use buffer state to filter out lazy buffers? I wonder if we need a new buffer state then, like `BUFFERS_ADDED` (similar to `SourcePartitionedScheduler.State#SPLITS_ADDED`). Then we could maybe default to `0` for lazy buffer utilization?

I do think defaulting to `0` for non initialized lazy buffer is clearer.

Reply:
Yes, it would require a new buffer state as you mentioned. To avoid that, using the buffer-utilization's presence (utilization = OptionalDouble) can be a solution. But if it is becoming unclear, I can maybe add another state - although I think the optional field for utilization would still be needed incase anyone just uses the utilization param.

Reply:
> But if it is becoming unclear, I can maybe add another state - although I think the optional field for utilization would still be needed incase anyone just uses the utilization param.

I think `OptionalDouble` is fine for now. However, it should be documented when exactly it can be empty.